src/many-abci/src/module.rs

```diff
@@ -13,32 +12,44 @@ use many_types::blockchain::{
 };
 use many_types::{blockchain::RangeBlockQuery, SortOrder, Timestamp};
 use once_cell::sync::Lazy;
+use sha2::Digest;
 use std::ops::{Bound, RangeBounds};
 use tendermint::Time;
-use tendermint_rpc::{query::Query, Client};
+use tendermint_rpc::{query::Query, Client, Order};
 
 const MAXIMUM_BLOCK_COUNT: u64 = 100;
 static DEFAULT_BLOCK_LIST_QUERY: Lazy<Query> = Lazy::new(|| Query::gte("block.height", 0));
 
-fn _many_block_from_tendermint_block(block: tendermint::Block) -> Block {
+fn _many_block_from_tendermint_block<C: Client + Sync>(
+    block: tendermint::Block,
+    args: blockchain::ListArgs,
+    client: &C,
+) -> Result<Block, ManyError> {
+    let (count, order, query) = transform_list_args(args)?;
+    let transaction_results_by_id = tx_results(client, count, order, query)?;
```

Comment:
You're fetching the last 100 transactions here. There's no guarantee that the requested `block` will be present in the result.

Reply:
@fmorency Isn't what transactions are requested determined by the `RangeBlockQuery` submitted?  In other words, aren't we asking for transactions that are within a block range of `tx_search`?